### ISSUE TITLE
Support adding an arbitrary reaction

### ISSUE THREAD

**Issue by gnprice** [zulip/zulip-flutter#388]
This will be a followup to:
* #125 

After that issue (and for Beta 1), we'll have the ability to add a thumbs-up from the message action sheet, and to tap on an existing reaction to join in or to remove it. This issue is to add the ability to choose an arbitrary reaction to add.

The zulip-mobile RN app has this functionality but the UI is kind of clumsy. With Flutter we have a lot more control to make the UI work however we like. So this is an opportunity to reimagine this UI to be much nicer. /cc @alya @terpimost

Some previous threads with ideas for improving this UI in the context of the zulip-mobile RN app:
* https://github.com/zulip/zulip-mobile/issues/4565 (and [chat thread](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/Quick.20reactions/near/1156419))
* https://github.com/zulip/zulip-mobile/issues/4567
* https://github.com/zulip/zulip-mobile/issues/4578

I think all three of those are good ideas, all together or in any combination, or we can go in another direction if we think it'd be good. For the third one (a button next to existing reactions) we can borrow the design from web, but the other two don't correspond to anything in web and so we'd need fresh designs for them.

**Comment by chrisbobbe** [comment]
A user reports in [CZO](https://chat.zulip.org/#narrow/stream/48-mobile/topic/order.20of.20emojis.20for.20reaction/near/1706954) that zulip-mobile's choice of putting `:heart:`, `:love:`, and `:love_you:` at the top of the emoji picker doesn't make sense for a work context: "I’m always a bit scared I’ll tap by mistake on the love-you emoji which is pretty close." Which makes sense; I've had a similar feeling myself! 😅

**Comment by chrisbobbe** [comment]
We've had another report by email, of wanting a less clumsy add-reaction UI than we currently have in zulip-mobile. They mention (if I understand correctly) having some frequently-used emoji directly in the message-long-press sheet, as discussed here.

They also had another interesting idea:

> If I can configure a swipe left action to be a certain emoji I would love that I think

**Comment by alainjr10** [comment]
Hey. I started doing some prelimary contributions to this issue, however, i can't seem to be able to pull all the available emojis. According to #669 , there's still an open issue about tracking all supported emojis. However, on Zulip web client, we can access a list of all emojis. so i was wondering how i can access those emojis to use on mobile.

**Comment by gnprice** [comment]
Yeah, as the issue description says, #669 is a prerequisite for this issue. So is #90. Both of those are in progress, but this issue won't be ripe to work on until those are done.

To see more about how one can access the emoji on mobile, take a look at #669 and the PRs linked from it — those are the code we wrote as the answer to that question :slightly_smiling_face:
